src/java/za/co/mip/ablduck/ABLDuck.java

```diff
@@ -0,0 +1,463 @@
+
+            if (stream == null) {
+                throw new IOException(
+                        "Cannot get resource \"" + resourceName + "\" from Jar file.");
+            }
+
+            int readBytes;
+            byte[] buffer = new byte[BUFFER_SIZE];
+
+            while ((readBytes = stream.read(buffer)) > 0) {
+                resStreamOut.write(buffer, 0, readBytes);
+            }
+        } catch (IOException ex) {
+            throw ex;
+        }
+
+        return resource;
+    }
+
+    public void unzip(File zipFile, File unzipTo) throws IOException {
+
+        if (!unzipTo.exists()) {
+            unzipTo.mkdir();
+        }
+
+        try (ZipInputStream zipIn = new ZipInputStream(
+                new FileInputStream(zipFile.getAbsolutePath()))) {
+            ZipEntry entry = zipIn.getNextEntry();
+
+            while (entry != null) {
+                String filePath = Paths.get(unzipTo.getAbsolutePath(), entry.getName()).toString();
+                if (!entry.isDirectory()) {
+                    extractFile(zipIn, filePath);
+                } else {
+                    File dir = new File(filePath);
+                    dir.mkdir();
+                }
+                zipIn.closeEntry();
+                entry = zipIn.getNextEntry();
+            }
+        } catch (IOException ex) {
+            throw ex;
+        }
+    }
+
+    private static void extractFile(ZipInputStream zipIn, String filePath) throws IOException {
+
+        try (BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(filePath))) {
+            byte[] bytesIn = new byte[BUFFER_SIZE];
+            int read = 0;
+            while ((read = zipIn.read(bytesIn)) != -1) {
+                bos.write(bytesIn, 0, read);
+            }
+        } catch (IOException ex) {
+            throw ex;
+        }
+    }
+
+    private void replaceTemplateTags(String tag, String value, java.nio.file.Path file)
+            throws IOException {
+        Charset charset = StandardCharsets.UTF_8;
+
+        String content = new String(Files.readAllBytes(file), charset);
+        content = content.replaceAll(Pattern.quote(tag), Matcher.quoteReplacement(value));
+        Files.write(file, content.getBytes(charset));
+    }
+
+    public static String determineUsingClass(Map<String, SourceJSObject> classes,
+            SourceJSObject cls, String dataType) {
+        if (classes.get(dataType) != null)
+            return dataType;
+
+        // First check if we have a direct using statement for the datatype
+        for (Using using : cls.using) {
+            String[] u = using.name.split("\\.");
+            if (u[u.length - 1].equals(dataType) && classes.get(using.name) != null) {
+                return using.name;
+            }
+        }
+
+        // Now check for * using statements
+        for (Using using : cls.using) {
+            String[] u = using.name.split("\\.");
+            if (u[u.length - 1].equals("*")) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "*" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1132)